src/main/java/net/unaussprechlich/warlordsplus/module/modules/FlagTakenDetector.kt

```diff
@@ -0,0 +1,43 @@
+package net.unaussprechlich.warlordsplus.module.modules
+
+import net.minecraft.client.Minecraft
+import net.minecraftforge.client.event.ClientChatReceivedEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import net.unaussprechlich.eventbus.EventBus
+import net.unaussprechlich.eventbus.IEvent
+import net.unaussprechlich.warlordsplus.module.IModule
+import net.unaussprechlich.warlordsplus.util.contain
+import net.unaussprechlich.warlordsplus.util.removeFormatting
+
+object FlagTakenDetector : IModule {
+
+    @SubscribeEvent
+    fun onChatMessage(e: ClientChatReceivedEvent) {
+        if (GameStateManager.notIngame || e.type == 2.toByte()) return
+        try {
+            val textMessage: String = e.message.unformattedText.removeFormatting()
+            if (textMessage.contains("picked up the")) {
```

Comment:
use `val player = when  { ... -> ... }` instead of the if/else spam

Reply:
4 lines weirdo